pco/src/data_types/mod.rs

```diff
@@ -16,6 +16,8 @@ mod floats;
 mod signeds;
 mod unsigneds;
 
+pub(crate) type ModeAndLatents<L> = (Mode<L>, Vec<Vec<L>>);
```

Comment:
perhaps document what the `Vec<Vec<L>>` means?  Something like
```suggestion
/// Container type for (i) a mode, and (ii) the result of transforming a dataset using that mode
/// (see `split_latents`).  The number of elements in the outer `Vec` equals the number of latents
/// that each (float) element of the dataset is split into, and the length of each inner `Vec` is
/// the number of elements in the dataset.
pub(crate) type ModeAndLatents<L> = (Mode<L>, Vec<Vec<L>>);
```

Reply:
Since this is pub(crate) it won't be exposed to users - they'll see the the full type instead of the alias. How about we document `choose_mode_and_split_latents` instead?

Reply:
Ah yeah, typedefs in general are hard to document for this reason, true.  Adding a sentence like the above to the doc of `choose_mode_and_split_latents` sounds good.

Reply:
Done